src/common/utils.js

```diff
@@ -123,13 +123,23 @@ const iconWithLabel = (icon, label, testid, iconSize) => {
 /**
  * Retrieves num with suffix k(thousands) precise to 1 decimal if greater than 999.
  *
- * @param {number} num The number to format.
- * @returns {string|number} The formatted number.
+ *  num The number to format.
+ *  The formatted number.
  */
 const kFormatter = (num) => {
-  return Math.abs(num) > 999
+  /* return Math.abs(num) > 999
     ? Math.sign(num) * parseFloat((Math.abs(num) / 1000).toFixed(1)) + "k"
-    : Math.sign(num) * Math.abs(num);
+    : Math.sign(num) * Math.abs(num); */
+
+  if (typeof num == "number") {
```

Comment:
Why does this need to be changed? This function should only accept numbers. In its current form strings should be converted before they enter the function.

Reply:
I changed this because while rendering the streaks card, it requires some additional information in the value (like - `days` or `weeks`) <br>
![image](https://github.com/anuraghazra/github-readme-stats/assets/129659514/80b53149-35b5-43aa-b430-dd33d6aad63f)
If this is reverted to the original code, then it will show `NaN` in the value instead of `2 weeks` or something.
This change will not affect the other cards. But, if you want, then I can make a separate function in the `utils.js` specifically for the `Streaks card`?

Reply:
I see. You should handle this problem outside the kFormatter function. Maybe only use the kformatter function when the value is a number:

https://github.com/anuraghazra/github-readme-stats/blob/33f132bd8c89701af2e91aa0a34f4601c65cf580/src/cards/streak-card.js#L19